logstash-core/lib/logstash/agent.rb

```diff
@@ -10,80 +10,125 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :logger
+  attr_reader :logger, :state
 
-  def initialize(logger)
-    @logger = logger
+  def initialize(params)
+    @logger = params[:logger]
     @pipelines = {}
+    @pipeline_threads = {}
+    @state = clean_state
```

Comment:
I think a comment helps. Ideally that comment would include the contents of your reply here and mention that other agents might use something richer.